src/extension.ts

```diff
@@ -187,23 +188,36 @@ function initCommand(context: vscode.ExtensionContext, commandId: string, callba
 function initAsyncCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => Promise<any>) {
     context.subscriptions.push(vscode.commands.registerCommand(commandId, async (...args: any[]) => {
         const start = Date.now();
+        const properties: { [key: string]: string; } = {};
         let result = 'Succeeded';
-        let errorData: string = '';
+        let errorData: ErrorData | undefined;
 
         try {
             await callback(...args);
         } catch (err) {
+            if (err instanceof LocalGitDeployError) {
+                properties.servicePlan = err.servicePlanSize;
+            }
             if (err instanceof UserCancelledError) {
                 result = 'Canceled';
+                errorData = new ErrorData(err);
+            } else if (err instanceof GitNotInstalledError) {
+                result = 'Failed';
+                errorData = new ErrorData(err);
             } else {
                 result = 'Failed';
-                errorData = util.errToString(err);
-                vscode.window.showErrorMessage(errorData);
+                errorData = new ErrorData(err);
+                vscode.window.showErrorMessage(errorData.message);
                 throw err;
```

Comment:
Don't throw 'err' here. We don't want to risk VS Code showing the error message again